boards/common/nrf52/dist/openocd.cfg

```diff
@@ -1,3 +1,6 @@
-transport select swd
```

Comment:
Maybe this line could simply be moved to https://github.com/RIOT-OS/RIOT/blob/master/makefiles/tools/openocd-adapters/jlink.inc.mk

With a line like:
```mk
OPENOCD_ADAPTER_INIT ?= -c 'transport select swd'
```

Other boards like kw41z which can also be flashed using openocd and the jlink adapter could be simplified the same wayl.
Another benefit is that one could also use `DEBUG_ADAPTER=dap` and a DAPLink programmer without any change in the code to be able to flash the nrf52 boards.

Reply:
Any approach that is better than mine is welcome. I'm not familiar with J-Link or any other adapters. The only adapter I have ever used is ST-LINKV2 with nucleo-64 boards. The change I suggested was simply the way I figured out how to use my ST-LINKV2 for the nRF52 board in PR #13860 because it doesn't have a J-Link adapter on the board.

https://github.com/RIOT-OS/RIOT/blob/b4655f6c91201dba3cb24c2ed201a2ef8741df32/makefiles/tools/openocd-adapters/jlink.inc.mk#L2
should probably be changed to 
```
OPENOCD_ADAPTER_INIT ?= -c 'source [find interface/jlink.cfg]' -c 'transport select swd'
```
Right?

I'm just wondering whether all boards that use OpenOCD and J-Link adapter also use SWD.

Reply:
> Right ?

You got my point :)

> I'm just wondering whether all boards that use OpenOCD and J-Link adapter also use SWD.

Good question. In fact, Jlink adapter supports both SWD and JTAG (see [here](http://openocd.org/doc/html/Debug-Adapter-Configuration.html)).
But in RIOT, all boards flashed using jlink are using SWD, for the moment. We could have a new make variable for that `OPENOCD_JLINK_ADAPTER_TRANSPORT` or something like that set by default to `swd` and handle the possible values (swd/jtag) in jlink.inc.mk. What do you think ?

Reply:
> Jlink adapter supports both SWD and JTAG

Therefore I was wondering whether all boards use SWD as transport protocol.

> We could have a new make variable for that `OPENOCD_JLINK_ADAPTER_TRANSPORT` or something like that set by default to `swd` and handle the possible values (swd/jtag) in `jlink.inc.mk`. What do you think ?

This is the way as it is done in `stlink.inc.mk`. The question is how we can figure out which boards use or could use `jlink` and which of them set a transport protocol other than `swd`, which would then has to be set in the board's `Makefile.include`. If we suppose that a board could use `openocd` has to define an `openocd.cfg` file, the following boards select a certain transport protocol:
```
boards/mcb2388/dist/openocd.cfg                 jtag
boards/msba2/dist/openocd.cfg                   jtag
boards/ek-lm4f120xl/dist/openocd.cfg            hla_jtag       
boards/common/nrf52/dist/openocd.cfg            swd
boards/common/cc26x2_cc13x2/dist/openocd.cfg    jtag
boards/common/frdm/dist/openocd-klx.cfg         swd
```
Some of them don't use `jlink` but `dap`.

Reply:
Most of those don't use `DEBUG_ADAPTER` though, which means the Makefiles in `openocd-adapters/` do no get included.